main/webapp/modules/core/scripts/project/history-panel.js

```diff
@@ -288,9 +288,32 @@ HistoryPanel.prototype._showApplyOperationsDialog = function() {
   
   elmts.dialogHeader.html($.i18n('core-project/apply-operation'));
   elmts.or_proj_pasteJson.html($.i18n('core-project/paste-json'));
+
+  elmts.operationJsonButton.on('click', async function() {
+    const fileInput = elmts.operationJsonButton[0];
+    fileInput.accept = '.json';
+    fileInput.onchange = async function() {
+      const file = fileInput.files[0];
+      const reader = new FileReader();
+      reader.onload = function(e) {
+        const fileContent = JSON.parse(e.target.result);
+        const textAreaElement = elmts.textarea[0];
+        if (textAreaElement) {
+          textAreaElement.textContent = JSON.stringify(fileContent, null, 2)
+        }
+      };
+      reader.addEventListener('error', function() {
+        alert('Error: Failed to read file');
```

Comment:
Hi @IjayAbby, as mentioned earlier this would still need localizing before we can merge this.

Reply:
Hello @wetneb, how do I do that Open Refine has many languages to be translated under? Kindly advise
This is my approach 
First, I need to include the jquery.i18n library in the project and initialize it with the appropriate translation files for the languages I want to support.
```
$.i18n().load({
  'en': 'path/to/en.json',
  'fr': 'path/to/fr.json'
});
```
Then, I can use the $.i18n() function to retrieve the localized string for the error message:

```
reader.addEventListener('error', function() {
  alert($.i18n('error_message'));
});
```

Reply:
Your first step is not necessary because the translation files are already loaded (to translate the rest of the app). So you can directly move on to your second step.

Reply:
Done.